### PR TITLE
Countdown for nearest tournament

### PR DESCRIPTION
**Pull request by Paryz**
added tournament countdown

image

### CODE REVIEW

File: test/eliscore_web/controllers/page_controller_test.exs [hunk 1]
```diff
@@ -1,7 +1,12 @@
 defmodule EliscoreWeb.PageControllerTest do
   use EliscoreWeb.ConnCase
+  alias Eliscore.{Repo, Tournament}
 
   test "GET /", %{conn: conn} do
+    Tournament.changeset(
+      %Tournament{},%{start_date: "2100-01-01 12:00:00"}
```
Comment (siepet) [inline test/eliscore_web/controllers/page_controller_test.exs]: missing space after `,`

File: assets/js/actions/index.js [hunk 2]
```diff
@@ -92,6 +100,19 @@ export const fetchCategories = () => {
   }
 }
 
+export const fetchTournamentDate = () => {
+  return dispatch => {
+    return axios.get("/api/v1/tournaments/date", { headers: buildHeaders() })
+      .then((response) => {
+
```
Comment (siepet) [inline assets/js/actions/index.js]: Empy line

File: assets/js/components/timer.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+  componentWillUnmount () {
+    window.clearInterval(this.interval)
+  }
+
+  showDate (difference, d, h, m, s) {
+    if (difference > 0) {
+      return `${d}d ${h}h ${m}m ${s}s left untill next TOURNAMENT`
+    } else {
+      return 'EXPIRED'
```
Comment (siepet) [inline assets/js/components/timer.js]: What about something more meaningful, example: `No tournaments scheduled`? 🤔

File: assets/js/components/timer.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    }
+  }
+
+  render () {
+    const date = new Date(this.state.date)
+    const startDate = new Date(this.props.startDate)
+    const difference = Math.floor(startDate.getTime() - date.getTime())
+    let {days, hours, minutes, seconds} = moment.duration(difference)._data
```
Comment (siepet) [inline assets/js/components/timer.js]: missing spaces in brackets

File: assets/js/components/header.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+import { Link } from 'react-router'
+import { connect } from 'react-redux'
+import { fetchTournamentDate } from '../actions/index'
+import Timer from './timer.js'
+
+class Header extends React.Component {
+  componentWillMount () {
+    const { dispatch } = this.props
+    dispatch(fetchTournamentDate())
+  }
```
Comment (virzen) [inline assets/js/components/header.js]: Consider using `mapDispatchToProps` for this.

**Review by siepet — APPROVED** [review]

File: assets/js/components/timer.js (hunk 4 above)
Comment (virzen) [inline assets/js/components/timer.js]: You can actually let moment calculate the difference and the format it as you wish:
- diffing https://stackoverflow.com/a/25150793/5670873,
- formatting https://momentjs.com/docs/#/displaying/format/.

Comment (virzen) [reply assets/js/components/timer.js]: Disclaimer: I didn't know that before. 🙈

**Pull request merged**